0xPolygon/polygon-edge
# Added new server config flag `--bootnode-only-sync` to ignore non-bootnodes in BestPeer()

# Description

When testing potential breaking changes on our testnets, we occasionally must stop all nodes and revert to a recent backup in order to revert the blockchain state to what it was prior to the breaking change. This process is fairly straightforward while dealing with a chain entirely operated by us. 

The issue lies when other peers or validators are connected to our mainnet chain which we do not have control over. As a precaution, if a new version of edge containing a breaking change was introduced on mainnet, it would be difficult to perform any kind of revert as our bootnodes would always sync to a peer that could have a higher blockNumber than the backup we want to revert to.

This PR introduces a new flag `--bootnode-only-sync` that we could require all validators to add after clearing their data-dirs to facilitate the unlikely event of having to reset to an earlier state. Additionally we would also add this flag to our bootnodes.

# Changes include


# Checklist


## Testing


### Manual tests

1. Spin up 4-node testnet with all 4 nodes as bootnodes
2. Add a 5th node to the testnet and give enough time to sync up
3. Clear the data (`rm -rf data/blockchain/ data/trie/ data/consensus/snapshots data/consensus/metadata`) of node 1.
4. Restart node 1 with `./main server --bootnode-only-sync` and observe that the non-bootnode peer is ignored when determining best peer within BestPeer()

# Documentation update

Will create a documentation PR once this PR is accepted.
